build/maven.go

```diff
@@ -174,43 +180,86 @@ func (mm *MavenModule) loadMavenHome() (mavenHome string, err error) {
 		// Since Maven installation can be located in different locations,
 		// Depending on the installation type and the OS (for example: For Mac with brew install: /usr/local/Cellar/maven/{version}/libexec or Ubuntu with debian: /usr/share/maven),
 		// We need to grab the location using the mvn --version command
-
 		// First we will try lo look for 'mvn' in PATH.
-		mvnPath, err := exec.LookPath("mvn")
-		if err != nil || mvnPath == "" {
-			return "", errors.New(err.Error() + "Hint: The mvn command may not be included in the PATH. Either add it to the path, or set the M2_HOME environment variable value to the maven installation directory, which is the directory which includes the bin and lib directories.")
+		maven, err := mm.getExecutableName()
+		if err != nil {
+			return maven, err
 		}
-		mm.containingBuild.logger.Debug(MavenHome, " is not defined. Retrieving Maven home using 'mvn --version' command.")
-		cmd := exec.Command("mvn", "--version")
-		var stdout bytes.Buffer
-		cmd.Stdout = &stdout
-		err = cmd.Run()
+		if !mm.extractorDetails.useWrapper {
+			mvnPath, err := exec.LookPath("mvn")
+			err = mm.determineError(mvnPath, "", err)
```

Comment:
We should call the `determineError` function from inside the functions that use it, becuase it is responsible for creating the error the function returns.